pkg/computenodeopenstack/machineset.go

```diff
@@ -0,0 +1,46 @@
+package computenodeopenstack
+
+import (
+	"context"
+
+	"github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	computenodev1alpha1 "github.com/openstack-k8s-operators/compute-node-operator/pkg/apis/computenode/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetMachineSetsWithLabel - Returns list of machinesets labeled with labelSelector
+func GetMachineSetsWithLabel(c client.Client, instance *computenodev1alpha1.ComputeNodeOpenStack, labelSelector map[string]string, namespace string) ([]*v1beta1.MachineSet, error) {
+	machineSets := &v1beta1.MachineSetList{}
+	if err := c.List(context.Background(), machineSets, client.InNamespace(namespace)); err != nil {
```

Comment:
can we list already with the label filtering?

Reply:
yes, done